controlplane/api/v1beta1/conversion.go

```diff
@@ -0,0 +1,89 @@
+		return fmt.Errorf("converting KThreesControlPlaneMachineTemplate field from v1beta1 to v1beta2: %w", err)
+	}
+	out.MachineTemplate.NodeDrainTimeout = in.NodeDrainTimeout
+	out.MachineTemplate.InfrastructureRef = in.InfrastructureTemplate
+	out.RemediationStrategy = (*cabp3v1.RemediationStrategy)(unsafe.Pointer(in.RemediationStrategy))
+	return nil
+}
+
+func Convert_v1beta2_KThreesControlPlaneSpec_To_v1beta1_KThreesControlPlaneSpec(in *cabp3v1.KThreesControlPlaneSpec, out *KThreesControlPlaneSpec, s conversion.Scope) error { //nolint: stylecheck
+	out.Replicas = in.Replicas
+	out.Version = in.Version
+	out.InfrastructureTemplate = in.MachineTemplate.InfrastructureRef
+	if err := Convert_v1beta2_KThreesConfigSpec_To_v1beta1_KThreesConfigSpec(&in.KThreesConfigSpec, &out.KThreesConfigSpec, s); err != nil {
+		return fmt.Errorf("converting KThreesConfigSpec field from v1beta2 to v1beta1: %w", err)
+	}
+	out.UpgradeAfter = in.UpgradeAfter
+	out.NodeDrainTimeout = in.MachineTemplate.NodeDrainTimeout
+	if err := Convert_v1beta2_KThreesControlPlaneMachineTemplate_To_v1beta1_KThreesControlPlaneMachineTemplate(&in.MachineTemplate, &out.MachineTemplate, s); err != nil {
+		return fmt.Errorf("converting KThreesControlPlaneMachineTemplate field from v1beta2 to v1beta1: %w", err)
+	}
+	out.RemediationStrategy = (*RemediationStrategy)(unsafe.Pointer(in.RemediationStrategy))
+	return nil
+}
+
+func Convert_v1beta2_KThreesControlPlaneMachineTemplate_To_v1beta1_KThreesControlPlaneMachineTemplate(in *cabp3v1.KThreesControlPlaneMachineTemplate, out *KThreesControlPlaneMachineTemplate, s conversion.Scope) error { //nolint: stylecheck
+	out.ObjectMeta = in.ObjectMeta
+	return nil
+}
+
+func Convert_v1beta1_KThreesConfigSpec_To_v1beta2_KThreesConfigSpec(in *bootstrapv1beta1.KThreesConfigSpec, out *bootstrapv1beta2.KThreesConfigSpec, s conversion.Scope) error { //nolint: stylecheck
+	return bootstrapv1beta1.Convert_v1beta1_KThreesConfigSpec_To_v1beta2_KThreesConfigSpec(in, out, s)
+}
+
+func Convert_v1beta2_KThreesConfigSpec_To_v1beta1_KThreesConfigSpec(in *bootstrapv1beta2.KThreesConfigSpec, out *bootstrapv1beta1.KThreesConfigSpec, s conversion.Scope) error { //nolint: stylecheck
+	return bootstrapv1beta1.Convert_v1beta2_KThreesConfigSpec_To_v1beta1_KThreesConfigSpec(in, out, s)
+}
+
+// ConvertTo converts the v1beta1 KThreesConfig receiver to a v1beta2 KThreesConfig.
+func (in *KThreesControlPlane) ConvertTo(dstRaw ctrlconversion.Hub) error {
+	dst := dstRaw.(*cabp3v1.KThreesControlPlane)
+	if err := Convert_v1beta1_KThreesControlPlane_To_v1beta2_KThreesControlPlane(in, dst, nil); err != nil {
+		return fmt.Errorf("converting KThreesControlPlane v1beta1 to v1beta2: %w", err)
+	}
+	return nil
+}
+
+// ConvertFrom converts the v1beta1 KThreesConfig receiver from a v1beta2 KThreesConfig.
+func (in *KThreesControlPlane) ConvertFrom(srcRaw ctrlconversion.Hub) error {
```

Comment:
If there are new fields on the new API version then you need to "store" the hub version by doing this:

```go
      if err := utilconversion.MarshalData(src, in); err != nil {
		return err
	}
```

As an example see [this](https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/main/controlplane/eks/api/v1beta1/conversion.go#L48).

Reply:
Wouldn't be a best practice to populate this annotation even if we are not making use of it currently?

Reply:
Nevermind, also breaks the fuzzy tests